src/IW/Core/Repo.hs

```diff
@@ -33,12 +36,35 @@ data Repo = Repo
     , repoName       :: !RepoName
     , repoDescr      :: !Text
     , repoCategories :: !(SqlArray Category)
+    , repoCabalUrl   :: !Url
     } deriving stock    (Eq, Generic, Show)
       deriving anyclass (ToJSON, FromRow, ToRow)
 
-repoUrl :: Repo -> Url
-repoUrl Repo{..} = Url
-    $ "https://github.com/"
-    <> unRepoOwner repoOwner
+-- | Convert a value of the @GitHub.Repo@ type to a value of our own @Repo@ type.
+fromGithubRepo :: GitHub.Repo -> Repo
+fromGithubRepo githubRepo = Repo
+    { repoOwner      = owner
+    , repoName       = name
+    , repoDescr      = fromMaybe "" $ GitHub.repoDescription githubRepo
+    , repoCategories = SqlArray []
+    , repoCabalUrl   = mkRepoCabalUrl owner name $ GitHub.repoDefaultBranch githubRepo
+    }
+  where
+    owner :: RepoOwner
+    owner = RepoOwner $ GitHub.untagName $ GitHub.simpleOwnerLogin $ GitHub.repoOwner githubRepo
+
+    name :: RepoName
+    name  = RepoName $ GitHub.untagName $ GitHub.repoName githubRepo
+
+-- | This function returns a @Url@ for downloading a @Repo@'s @.cabal@ file.
+mkRepoCabalUrl :: RepoOwner -> RepoName -> Maybe Text -> Url
+mkRepoCabalUrl (RepoOwner repoOwner) (RepoName repoName) maybeBranch = Url $
+    "https://raw.githubusercontent.com/"
+    <> repoOwner
+    <> "/"
```

Comment:
You can use `T.intercalate "/"` to not repeat slashes, but this is just the matter of taste :)